File: docs/vsto/codesnippet/VisualBasic/trin_excelworkbookpda/filecopypda/class1.vb
' <snippet7>
' <snippet1>
Imports Microsoft.VisualStudio.Tools.Applications.Deployment
Imports Microsoft.VisualStudio.Tools.Applications
' </snippet1>

' <snippet2>
Public Class FileCopyPDA
    Implements IAddInPostDeploymentAction
' </snippet2>

' <snippet3>
    Sub Execute(ByVal args As AddInPostDeploymentActionArgs) Implements IAddInPostDeploymentAction.Execute
        Dim dataDirectory As String = "Data\ExcelWorkbook.xlsx"
        Dim file As String = "ExcelWorkbook.xlsx"
        Dim sourcePath As String = args.AddInPath
        Dim deploymentManifestUri As Uri = args.ManifestLocation
        Dim destPath As String = Environment.GetFolderPath(Environment.SpecialFolder.DesktopDirectory)
        Dim sourceFile As String = System.IO.Path.Combine(sourcePath, dataDirectory)
        Dim destFile As String = System.IO.Path.Combine(destPath, file)

        Select Case args.InstallationStatus
            Case AddInInstallationStatus.InitialInstall, AddInInstallationStatus.Update
                System.IO.File.Copy(sourceFile, destFile)
                ServerDocument.RemoveCustomization(destFile)
                ServerDocument.AddCustomization(destFile, deploymentManifestUri)
                Exit Select
            Case AddInInstallationStatus.Uninstall
                If System.IO.File.Exists(destFile) Then
                    System.IO.File.Delete(destFile)
                End If
                Exit Select
        End Select
    End Sub
    ' </snippet3>
    ' </snippet7>

End Class

File: docs/vsto/codesnippet/CSharp/trin_excelworkbookpda/filecopypda/class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
// <snippet7>
// <snippet1>
using Microsoft.VisualStudio.Tools.Applications.Deployment;
using Microsoft.VisualStudio.Tools.Applications;
using System.IO;
// </snippet1>

namespace FileCopyPDA
{
// <snippet2>
    public class FileCopyPDA : IAddInPostDeploymentAction
// </snippet2>
    {
// <snippet3>
        public void Execute(AddInPostDeploymentActionArgs args) 
        {
            string dataDirectory = @"Data\ExcelWorkbook.xlsx";
            string file = @"ExcelWorkbook.xlsx";
            string sourcePath = args.AddInPath;
            Uri deploymentManifestUri = args.ManifestLocation;
            string destPath = Environment.GetFolderPath(Environment.SpecialFolder.DesktopDirectory);
            string sourceFile = System.IO.Path.Combine(sourcePath, dataDirectory);
            string destFile = System.IO.Path.Combine(destPath, file);

            switch (args.InstallationStatus)
            {
                case AddInInstallationStatus.InitialInstall:
                case AddInInstallationStatus.Update:
                    File.Copy(sourceFile, destFile);
                    ServerDocument.RemoveCustomization(destFile);
                    ServerDocument.AddCustomization(destFile, deploymentManifestUri);
                    break;
                case AddInInstallationStatus.Uninstall:
                    if (File.Exists(destFile))
                    {
                        File.Delete(destFile);
                    }
                    break;
            }
        }
// </snippet3>
    }
}
// </snippet7>
